analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1260,6 +1255,53 @@ public void Invocation_BoolEquals_BothArgumentsHaveNoConstraint_DoesNotLearn()
         validator.TagValue("Result").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
 
+    [DataTestMethod]
+    [DataRow("42", "42", true)]
+    [DataRow("0", "42", false)]
+    [DataRow("((int?)42)", "((int?)42)", true)]
+    [DataRow("((int?)42)", "((int?)0)", false)]
+    public void Invocation_NumberEquals_SingleValues_LearnsResult(string left, string right, bool expected)
+    {
+        var code = $"""
+                var result = {left}.Equals({right});
+                Tag("Result", result);
+
+                var resultStaticCall = object.Equals({left}, {right});
+                Tag("ResultStaticCall", resultStaticCall);
+                """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+        validator.TagValue("ResultStaticCall").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+    }
+
+    [TestMethod]
+    public void Invocation_NumberEquals_DoesNotLearn()
```

Comment:
Needs to explain how is it different from the others
```suggestion
    public void Invocation_NumberEquals_ValueInRange_DoesNotLearn()
```